grouper/service_account.py

```diff
@@ -0,0 +1,163 @@
+from grouper.constants import USER_ADMIN
+from grouper.group import user_can_manage_group
+from grouper.model_soup import Group
+from grouper.models.user import User
+from grouper.user import disable_user, enable_user
+from grouper.user_permissions import user_has_permission
+
+
+class ServiceAccountNotFound(Exception):
+    pass
+
+
+def is_service_account(session, user=None, group=None):
+    # type: Session, User, Group -> bool
+    """
+    Takes in a User or a Group and returns a boolean indicating whether
+    that User/Group is a component of a service account.
+
+    Args:
+        session: the database session
+        user: a User object to check
+        group: a Group object to check
+
+    Returns:
+        whether the User/Group is a component of a service account
+    """
+    if not user and not group:
+        return False
+
+    if user is not None:
+        return user.role_user
+
+    user = User.get(session, name=group.groupname)
+    if not user:
+        return False
+
+    return user.role_user
+
+
+def get_service_account(session, user=None, group=None):
+    # type: Session, User, Group -> Dict[str, Union[User, Group]]
```

Comment:
only a thought: might be worth making a `namedtuple` that is a representation of a service account. slightly stronger 'typing' to prevent future crap tagging along.